Add methods to set AddrPort from NodeItem and convert back

gavin-norman-sociomantic opened the pull request:
As part of replacing internal usages of NodeItem with AddrPort, it's useful
to have a convenient way of converting from one to the other.

Part of #348.

nemanja-boric-sociomantic commented on src/swarm/neo/AddrPort.d:
```diff
@@ -25,11 +25,15 @@ public struct AddrPort
     import swarm.util.Verify;
 
+    /// Minimum length required for an address format buffer.
+    public static const AddrBufLength = INET_ADDRSTRLEN;
```
I would play it safe here and I would go with `INET6_ADDRSTRLEN`.

gavin-norman-sociomantic replied:
Updated.

Commit:
Add methods to set AddrPort from NodeItem and convert back

As part of replacing internal usages of NodeItem with AddrPort, it's useful
to have a convenient way of converting from one to the other.

Part of #348.

Geod24 commented on src/swarm/neo/AddrPort.d:
```diff
@@ -25,11 +25,15 @@ public struct AddrPort
     import swarm.util.Verify;
 
+    /// Minimum length required for an address format buffer.
+    public static const AddrBufLength = INET6_ADDRSTRLEN;
```
`static` is redundant (and wouldn't compile in a `class`)

gavin-norman-sociomantic replied:
:+1: I'm forgetting what works in D1 and D2. Is it just `const`?

nemanja-boric-sociomantic replied:
Yup, just `const` works.

Geod24 replied:
`const` + initializer

nemanja-boric-sociomantic replied:
Ping @gavin-norman-sociomantic :-)

gavin-norman-sociomantic replied:
Updated.
